examples/test_modified_pipeline.py

```diff
@@ -0,0 +1,19 @@
+from pathlib import Path
+from typing import Iterator
+
+from mat_dp_pipeline import Pipeline
+from mat_dp_pipeline.pipeline import LabelledOutput, ProcessableFullType
+
+
+class ModifiedPipeline(Pipeline):
+    def _processable_to_processed(
+        self, processable: Iterator[Iterator[ProcessableFullType]]
+    ) -> Iterator[LabelledOutput]:
+        out = super()._processable_to_processed(processable)
+        print(list(out))
+        return out
+
+
+pipeline = ModifiedPipeline()
+
+output = pipeline(Path("./World"))
```

Comment:
How about `world = Path(__file__).parent.parent / "tests/data/World"` ? Alternatively have some example_utils which will give you the path to tests/data/. I'd also vote for renaming the files to 1) either remove test_ prefix if they are not tests or 2) move examples/ to tests/examples if they are tests (maybe not tests per se, but something which could be always run when we run the tests, more like a sanity check tests)

Reply:
Yeah 1) seems like a good shout

Reply:
Ooh new idea

Reply:
New directory - test_data. examples and tests share this directory

Reply:
I want to keep examples and tests separate so that you can run `poetry run pytest` and it'll only scan automated tests. I've had it before where depending on your example it can get confused otherwise

Reply:
Moving to a new directory sounds ok to me. I tend to run `pytest` from the main directory, it just looks for everything matching pytest rules. That's why I don't like `test_` if they are not tests and your are not willing for them to be executed as tests (which I'm not saying is a bad idea - it might be a good idea)

Reply:
ooh that's quite spicy - pytest looks for config files (e.g. pytest.ini) in the directory you run it, so launching from the tests directory I believe is also the convention

Reply:
https://docs.pytest.org/en/7.1.x/reference/customize.html#configuration-file-formats

I stand corrected but I think it's odd!